examples/twittermap/web/public/javascripts/mapresultcache/services.js

```diff
@@ -35,55 +51,127 @@ angular.module('cloudberry.mapresultcache', ['cloudberry.common'])
             // same length as the geoIds parameter in case of complete cache miss,
             // otherwise in range (0, geoIds.length)
             var geoIdsNotInCache = [];
-
+            
             // New query case
-            if (keywords.toString() != currentKeywords.toString() ||
-                !angular.equals(currentTimeRange, timeInterval)) {
+            if (keywords.toString() !== currentKeywords.toString()) {
                 store.clear();
+                countyStore.clear();
                 currentKeywords = keywords.slice();
                 currentTimeRange.start = timeInterval.start;
                 currentTimeRange.end = timeInterval.end;
+                return geoIds; // since cache is emptied.
+            } 
+            // County level case
+            if (geoLevel === "county") {
+                return [];
             }
```

Comment:
Do we always have data for all counties when geoLevel is county? I think it is better not to make such assumption. It is possible that the previous query only includes some of the counties.